### PR TITLE
Add grep tools - bingrep, ugrep, ripgrep-all

### PR DESCRIPTION
**Pull request by danielnachun**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by danielnachun** [comment]
@conda-forge/staged-recipes This is now ready for review.

I had to skip the Windows build for two packages:
- `bzip3` (optional dependency of `ugrep`) does not seem to support MSVC, and my attempts to use MinGW weren't successful
- `ugrep` uses autotools to configure its build.  I tried to use the `autotools_clang_conda` package without success.  The project also provides `.vcxproj` files that could potentially use `msbuild` to build them, but seems to require vendoring all of the dependencies, which I don't think is a good idea.

File: recipes/bzip3/build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+set -o xtrace -o nounset -o pipefail -o errexit
+
+cmake -S . -B build \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_VERBOSE_MAKEFILE=ON \
+    -Wno-dev \
+    -DBUILD_TESTING=OFF \
+    ${CMAKE_ARGS}
+
+cmake --build build
```
Comment (xhochy) [inline recipes/bzip3/build.sh:12]:
```suggestion
cmake --build build -j$CPU_COUNT
```

File: recipes/ugrep/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - zlib
+  run:
+    - libbrotlidec
+    - bzip2
+    - bzip3
+    - lz4-c
+    - pcre2
+    - xz
+    - zstd
+    - libzlib
```
Comment (xhochy) [inline recipes/ugrep/meta.yaml:44]: All of these should come as a `run_exports`
```suggestion
```

File: recipes/ugrep/build.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -o xtrace -o nounset -o pipefail -o errexit
+
+./configure --enable-color \
+    --disable-silent-rules \
+    --disable-debug \
+    --disable-dependency-tracking \
+    --prefix=${PREFIX}
+make
```
Comment (xhochy) [inline recipes/ugrep/build.sh:10]:
```suggestion
make -j$CPU_COUNT
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by danielnachun** [comment]
The requested changes have been made - see the last two commits.

**Review by xhochy — APPROVED** [review]

**Pull request merged**